ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -801,3 +813,7 @@ data TraceReplayEvent blk =
         (ReplayStart blk) -- ^ the block at which this replay started
         (ReplayGoal blk)  -- ^ the block at the tip of the ImmutableDB
   deriving (Generic, Eq, Show)
+
+data BackingStoreSelector m where
+  LMDBBackingStore :: !FsPath -> BackingStoreSelector IO
```

Comment:
Ah ha, I was worrying about how to expose `IO` for LMDB C calls. I hadn't though of using the selector to expose `IO`. Nice and tidy! 👏

Reply:
Yeah, `MonadIO` is fine instead of `IO`. The other args make sense too.